unit_tests/test_decode_backtrace.py

```diff
@@ -121,21 +135,149 @@ def test_04_decode_backtraces_core(self):
 
         Setup.DECODING_QUEUE = queue.Queue()
         Setup.BACKTRACE_DECODING = True
+        self.open_event_file()
 
         self.monitor_node.start_decode_on_monitor_node_thread()
         self.node.system_log = os.path.join(os.path.dirname(__file__), 'test_data', 'system_core.log')
 
-        self.node._read_system_log_and_publish_events()
-
+        self.node._read_system_log_and_publish_events(start_from_beginning=True)
+        # Wait for events been published
+        time.sleep(2)
         self.monitor_node.termination_event.set()
         self.monitor_node.stop_task_threads()
 
-        events_file = open(EVENTS_PROCESSES['MainDevice'].raw_events_filename, 'r')
-        events = []
-        for line in events_file.readlines():
-            events.append(json.loads(line))
+        events = [json.loads(line) for line in self._event_file]
+        self._event_file.close()
+
+        self.assertTrue(events, "Events were not found")
 
         for event in events:
             if event.get('backtrace') and event.get('raw_backtrace'):
                 self.assertEqual(event['backtrace'].strip(),
                                  "addr2line -Cpife scylla_debug_info_file {}".format(' '.join(event['raw_backtrace'].split("\n"))))
+
+
+class TestCoredumpRetryingWithException(unittest.TestCase):
+
+    _event_file = None
+
+    @classmethod
+    def setUpClass(cls):
+        cls.temp_dir = tempfile.mkdtemp()
+        start_events_device(cls.temp_dir, timeout=5)
+
+        cls.node = DecodeDummyNode(name='test_node', parent_cluster=None,
+                                   base_logdir=cls.temp_dir, ssh_login_info=dict(key_file='~/.ssh/scylla-test'))
+        cls.node.init()
+        cls.node.remoter = DummyRemoteWithException()
+        cls.node.backtrace_lookup_period = 0
+        cls.node.get_core_pids_func_retrying = 2
+        cls.node.get_core_pids_func_sleep_timeout = 1
+        cls.node.system_log = os.path.join(os.path.dirname(__file__), 'test_data', 'system.log')
+
+    @classmethod
+    def tearDownClass(cls):
+        stop_events_device()
+        shutil.rmtree(cls.temp_dir, ignore_errors=True)
+
+    def setUp(self):
+        self.open_event_file()
+        self.node.remoter.run_number = 0
+
+    def tearDown(self) -> None:
+        self._event_file.close()
+
+    def open_event_file(self):
+        self._event_file = open(EVENTS_PROCESSES['MainDevice'].raw_events_filename, 'r')
+        self._event_file.seek(0, 2)
+
+    def find_last_event(self):
+        events = [json.loads(line) for line in self._event_file]
+        traceback = ""
+        if events:
+            last_event = events[-1]
+            if "traceback" in last_event:
+                print(f"Last event: {last_event['traceback']}")
+                traceback = last_event["traceback"].splitlines()[-1]
+        return traceback
+
+    def wait_for_all_exceptions_raised(self, exceptions_amount):
+        while self.node.remoter.run_number < exceptions_amount:
```

Comment:
Same. It's not a loop. And why we compare `run_number` with `exceptions_amount`?

Reply:
changed